### PR TITLE
Deploy GnosisSafe for E2E test

### PR DESCRIPTION
**Pull request by nlordell**
This PR changes the existing `smart_contract_order` E2E test to actually use a smart contract 🤣.

While it does not _techinically_ matter for `presign` orders, now that we have preliminary EIP-1271 support and I want to add an E2E test, we need to actually deploy an SC wallet that supports on-chain signatures.

### Test Plan

This is the plan.

### CODE REVIEW

File: crates/contracts/artifacts/ERC1271SignatureValidator.json [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-            "type": "bytes4"
-          }
-        ],
-        "stateMutability": "view",
-        "type": "function"
-      }
-    ]
-  }
\ No newline at end of file
+  "abi": [
```
Comment (nlordell) [inline crates/contracts/artifacts/ERC1271SignatureValidator.json:2]: Just formatting.

File: crates/e2e/tests/e2e/services.rs [hunk 2]
```diff
@@ -60,10 +60,39 @@ macro_rules! tx {
...
+}
+
 pub fn to_wei(base: u32) -> U256 {
     U256::from(base) * U256::from(10).pow(18.into())
 }
 
+pub fn safe_validator_signature(owner: H160) -> Bytes<Vec<u8>> {
+    let mut signature = vec![0; 65];
+    signature[12..32].copy_from_slice(owner.as_bytes());
+    signature[64] = 1;
```
Comment (MartinquaXD) [inline crates/e2e/tests/e2e/services.rs:92]: I don't know if the signature is initialized correctly.
Is there maybe some smart contract code you can link here which could explain that?

**Review by MartinquaXD — APPROVED** [review]
Looks reasonable.
Just some reference for the initialization for the signature would be nice if that makes sense.

**Review by vkgnosis — APPROVED** [review]

File: crates/e2e/tests/e2e/services.rs (hunk 2 above)
Comment (nlordell) [reply crates/e2e/tests/e2e/services.rs:102]: Added comment explaining what signature scheme we are using with links to the Safe docs and SC code that verifies it.

**Pull request merged**